Dockerfile

```diff
@@ -0,0 +1,49 @@
+FROM php:8.3-apache
+
+# Update and install necessary packages
+RUN apt-get update && \
+    apt-get install -y libpng-dev libjpeg-dev libfreetype6-dev libopencv-dev && \
+    docker-php-ext-install gd && \
+    apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# Configure Apache
+RUN echo "<VirtualHost *:80>\n\
+    DocumentRoot /var/www/html\n\
+    <Directory /var/www/html>\n\
+    Options Indexes FollowSymLinks\n\
+    AllowOverride All\n\
+    Require all granted\n\
+    </Directory>\n\
+    # Redirect from root to Introduction.html\n\
+    RewriteEngine On\n\
+    RewriteCond %{REQUEST_URI} ^/$\n\
+    RewriteRule ^/$ /Introduction.html [R=301,L]\n\
+    </VirtualHost>" > /etc/apache2/sites-available/000-default.conf
+
+# Copy application files
+COPY ./build/ /var/www/html/
+COPY ../compile-all.sh /
+
+# Make compile script executable and run it
+RUN chmod +x /compile-all.sh && /compile-all.sh
+
+# Enable Apache modules and configurations
+RUN a2enmod rewrite && \
+    echo "<Directory /var/www/html>\n\
+    Options Indexes FollowSymLinks\n\
+    AllowOverride All\n\
+    Require all granted\n\
+    </Directory>" > /etc/apache2/conf-available/exp.conf && \
+    a2enconf exp && \
+    echo "ServerName localhost" > /etc/apache2/conf-available/servername.conf && \
```

Comment:
Ideally, localhost should be changed to the deployment URL. However, ignoring this line might also work because we are serving only one website in this container. Should I update it to `cse19-iiith.vlabs.ac.in` or will we test the deployment somewhere else before migrating it to the original URL?

Reply:
I guess the routing will work since we are using a single website here. However, I am not clear about the role of ServerName.
If ServerName is relevant and must contain the actual deployment url, then this should be governed by an environment variable since it depends on the environment.